### PR TITLE
Update the SDK and msbuild package version

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/367 and https://github.com/NuGet/Home/issues/8952
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 
1. Update the SDK version for build and test to 5.0.100
2. Update the msbuild packages version from 16.6.0 to 16.8.0
3. Update the pack task package from 5.7.0 to 5.8.0
4. Update the cryptography packages version to 5.0.0
5. Add System.Text.Encodings.Web.dll and System.Text.Json.dll into .vsixignore, the two dlls are brought in by 
  Microsoft.Build/16.8.0  => System.Text.Json/4.7.0 => System.Text.Encodings.Web/4.7.0
6. Remove the temporary patching on deps.json from `MsbuildIntegrationTestFixture`.
7.Fix a test which failed as new BuildEngine7 property is brought in.

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  Engineering
Validation:

### CODE REVIEW

**Comment by heng-liu** [comment]
Test `TestPackageManager_UpgradePackageFor_TopParentProject_Success` keeps failing in this branch, but I couldn't repro when running locally.
This test also looks flaky in other branches, but it doesn't constantly fail.
Hi @zivkan, may I know if I could skip this test?
Hi @dtivel , FYI, as you're blocked by this PR.

File: build/config.props
Comment (kartheekp-ms) [inline build/config.props:40]: I think we can remove `OverrideCliBranchForTesting` and `LockSDKVersion` properties and only rely on `CliVersionForBuilding` and `CliBranchForTesting` properties. This should also fix https://github.com/NuGet/Home/issues/9712.

Comment (zivkan) [reply build/config.props:40]: As this PR is needed to unblock CI builds for the whole team, we'll do the suggestion in another PR.

**Review by zivkan — APPROVED** [review]

**Pull request merged**
